app/components/Documentation/components/Pagination/PaginationDocumentation.tsx

```diff
@@ -0,0 +1,109 @@
+import React from 'react';
+
+import { getIpfsHashFromBytes32 } from 'utils/ipfs';
+import createdHistory from 'createdHistory';
+import * as routes from 'routes-config';
+import useMediaQuery from 'hooks/useMediaQuery';
+import { getDataFromTree, getcurrentArrayIndex } from '../../helpers';
+import { DocumentationItemMenuType } from 'pages/Documentation/types';
+import ButtonPaginationDesktop from './ButtonPaginationDesktop';
+import ButtonPaginationMobile from './ButtonPaginationMobile';
+
+type PaginationDocumentationProps = {
+  documentationMenu: DocumentationItemMenuType;
+  id: string;
+  onClickPaginationArticleEditDocumentation?: () => void;
+};
+
+const PaginationDocumentation: React.FC<PaginationDocumentationProps> = ({
+  documentationMenu,
+  id,
+  onClickPaginationArticleEditDocumentation,
+}): JSX.Element => {
+  const isDesktop = useMediaQuery('(min-width: 577px)');
+  const treeArray = getDataFromTree(documentationMenu);
+
+  const currentArrayIndex = getcurrentArrayIndex(treeArray, id);
+
+  const isStartArticle = currentArrayIndex < 1;
+  const isLastArticle = currentArrayIndex === treeArray.length - 1;
+  const TITLE_LENGTH = isDesktop ? 26 : 50;
+  const TITLE_LAST_INDEX = isDesktop ? 25 : 49;
+  const TITLE_START_INDEX = 0;
+  const TITLE_END = '...';
+  const NEXT_TYPE_BUTTON = 'next';
+  const PREV_TYPE_BUTTON = 'prev';
+
+  const onClickPaginationArticle = (typeButton: 'prev' | 'next') => () => {
+    const stepDirection = typeButton === PREV_TYPE_BUTTON ? -1 : +1;
+    const arrayId = treeArray.find(
+      (item) => item?.treeIndex === currentArrayIndex + stepDirection,
+    )?.node.id;
+    const arrayTitle = treeArray.find(
+      (item) => item?.treeIndex === currentArrayIndex + stepDirection,
+    )?.node.title;
+    const ipfsHash = getIpfsHashFromBytes32(arrayId);
+    createdHistory.push(routes.documentation(ipfsHash, arrayTitle));
+  };
+
+  const getcurrentArrayTitle = (typeButton: 'prev' | 'next') => {
+    const stepDirection = typeButton === PREV_TYPE_BUTTON ? -1 : +1;
+    const title = treeArray?.find(
+      (item) => item?.treeIndex === currentArrayIndex + stepDirection,
+    )?.node.title;
+    return title?.length < TITLE_LENGTH
```

Comment:
why? why don't you use css for shortening?

Reply:
done: https://github.com/peeranha/peeranha-web/pull/1255/commits/303a47c06288eb7a7876b1135fe75d5894b3359f